Code/GraphMol/Descriptors/AUTOCORR3D.cpp

```diff
@@ -0,0 +1,242 @@
+
+   VectorXd We = getEigenVect(we);
+
+   std::vector<double>  wu = moldata3D.GetUn(numAtoms);
+
+   VectorXd Wu = getEigenVect(wu);
+
+   std::vector<double>  ws =  moldata3D.GetIState(mol);
+
+   VectorXd Ws = getEigenVect(ws);
+
+   std::vector<double> wr= moldata3D.GetRelativeRcov(mol);
+
+   VectorXd Wr = getEigenVect(wr);
+
+ 
+   MatrixXd Bi;
+   MatrixXd tmp;
+   double TDBmat[8][10];
+   double dtmp;
+
+  for (int i=0;i<10;i++){
+    double * Bimat = GetGeodesicMatrix(dist, i+1, numAtoms);
+    Map<MatrixXd> Bi(Bimat, numAtoms,numAtoms);
+    MatrixXd RBi=Bi.cwiseProduct(dm);
+    double Bicount=(double) Bi.sum();
+
+    tmp=Wu.transpose() * RBi * Wu / Bicount;
+    dtmp=(double)tmp(0);
+    if  (std::isnan(dtmp)) dtmp=0.0;
+    TDBmat[0][i]=dtmp;
+
+    tmp=Wm.transpose() * RBi * Wm / Bicount;
+    dtmp=(double)tmp(0);
+    if  (std::isnan(dtmp)) dtmp=0.0;
+    TDBmat[1][i]=dtmp;
+
+    tmp=Wv.transpose() * RBi * Wv / Bicount;
+    dtmp=(double)tmp(0);
+    if  (std::isnan(dtmp)) dtmp=0.0;
+    TDBmat[2][i]=dtmp;
+
+    tmp=We.transpose() * RBi * We / Bicount;
+    dtmp=(double)tmp(0);
+    if  (std::isnan(dtmp)) dtmp=0.0;
+    TDBmat[3][i]=dtmp;
+
+    tmp=Wp.transpose() * RBi * Wp / Bicount;
+    dtmp=(double)tmp(0);
+    if  (std::isnan(dtmp)) dtmp=0.0;
+    TDBmat[4][i]=dtmp;
+
+    tmp=Wi.transpose() * RBi * Wi / Bicount;
+    dtmp=(double)tmp(0);
+    if  (std::isnan(dtmp)) dtmp=0.0;
+    TDBmat[5][i]=dtmp;
+
+    tmp=Ws.transpose() * RBi * Ws / Bicount;
+    dtmp=(double)tmp(0);
+    if  (std::isnan(dtmp)) dtmp=0.0;
+    TDBmat[6][i]=dtmp;
+
+    tmp=Wr.transpose() * RBi * Wr / Bicount;
+    dtmp=(double)tmp(0);
+    if  (std::isnan(dtmp)) dtmp=0.0;
+    TDBmat[7][i]=dtmp;
+
+  }
+
+  // create the Output vector!
+    w= AppendDouble(w, TDBmat[0], 10, 0);
+    w= AppendDouble(w, TDBmat[1], 10, 10);
+    w= AppendDouble(w, TDBmat[2], 10, 20);
+    w= AppendDouble(w, TDBmat[3], 10, 30);
+    w= AppendDouble(w, TDBmat[4], 10, 40);
+    w= AppendDouble(w, TDBmat[5], 10, 50);
+    w= AppendDouble(w, TDBmat[6], 10, 60);
+    w= AppendDouble(w, TDBmat[7], 10, 70);
+
+    return w;
+
+}
+
+  double* Get3Dauto(double* dist3D,double* dist, int numAtoms, const ROMol& mol){
+
+      std::vector<std::string> AUTOCORRNAMES={"TDB01u","TDB02u","TDB03u","TDB04u","TDB05u","TDB06u","TDB07u","TDB08u","TDB09u","TDB10u","TDB01m","TDB02m","TDB03m","TDB04m","TDB05m","TDB06m","TDB07m","TDB08m","TDB09m","TDB10m","TDB01v","TDB02v","TDB03v","TDB04v","TDB05v","TDB06v","TDB07v","TDB08v","TDB09v","TDB10v","TDB01e","TDB02e","TDB03e","TDB04e","TDB05e","TDB06e","TDB07e","TDB08e","TDB09e","TDB10e","TDB01p","TDB02p","TDB03p","TDB04p","TDB05p","TDB06p","TDB07p","TDB08p","TDB09p","TDB10p","TDB01i","TDB02i","TDB03i","TDB04i","TDB05i","TDB06i","TDB07i","TDB08i","TDB09i","TDB10i","TDB01s","TDB02s","TDB03s","TDB04s","TDB05s","TDB06s","TDB07s","TDB08s","TDB09s","TDB10s","TDB01r","TDB02r","TDB03r","TDB04r","TDB05r","TDB06r","TDB07r","TDB08r","TDB09r","TDB10r"};
```

Comment:
I think it would be important to find a solution that links values with names. This is like a comment to the code (http://stackoverflow.com/questions/167498/what-is-less-annoying-no-source-code-documentation-or-bad-code-documentation).